battle.go

```diff
@@ -190,7 +207,7 @@ func (b *Battle) SimulateRound() ([]Transaction, bool) {
 					})
 					continue // forfeit turn
 				}
-			} else if user.StatusEffects.check(StatusSleep) {
+			} else if user.StatusEffects.check(StatusSleep) && move.ID != MoveSnore && move.ID != MoveSleepTalk {
 				b.QueueTransaction(ImmobilizeTransaction{
 					Target: target{
 						Pokemon: user,
```

Comment:
(meta) Since this is just gonna keep growing, it might be good to split up SimulateRound some more, like `b.getTurns() []TurnContext`, `b.sortTurns(turns *[]TurnContext)`. Probably don't need a separate function to process turns, we can keep that in `SimulateRound`.